arch/arc/mm/ioremap.c

```diff
@@ -110,7 +110,7 @@ EXPORT_SYMBOL(ioremap_prot);
 void iounmap(const void __iomem *addr)
 {
 	/* weird double cast to handle phys_addr_t > 32 bits */
-	if (arc_uncached_addr_space((phys_addr_t)addr))
+	if (arc_uncached_addr_space((unsigned long __force)addr))
```

Comment:
The previous diff above converts phys_addr_t to pointer. In this function we have to convert pointer to phys_addr_t. So why change phys_addr_t to pointer conversion here ?

Reply:
Because in case of ```sizeof(phys_addr_t) != sizeof(void *)``` we have conversion of pointer to integer of different size. So here we can write something like ```(phys_addr_t) (unsigned long) addr```, but since function argument is ```phys_addr_t``` I just converted ```addr``` to ```unsigned long``` (because ```sizeof(unsigned long) == sizeof(void *)```.